features/storage/blockdevice/MBRBlockDevice.h

```diff
@@ -52,21 +52,25 @@ class MBRBlockDevice : public BlockDevice {
      *
      *  @param bd       Block device to partition
      *  @param part     Partition to use, 1-4
-     *  @param type     8-bit partition type to identify partition contents
+     *  @param type     8-bit partition type or ID to identify partition contents
      *  @param start    Start block address to map to block 0 of partition.
      *                  Negative addresses are calculated from the end of the
      *                  underlying block devices. Block 0 is implicitly ignored
      *                  from the range to store the MBR.
      *  @return         0 on success or a negative error code on failure.
      *  @note This is the same as partition(bd, part, type, start, bd->size())
+     *  @note 8-bit partition type or ID is intended to specify the filesystem partition
+     *        contains and/or to flag special access methods used to access these partitions
+     *        for example like CHS mapping, LBA access etc.,
+     *        The list of Partition types or IDS refer https://en.wikipedia.org/wiki/Partition_type#List_of_partition_IDs
```

Comment:
The partition type gets stored in the MBR partition table, so it needs to match the actual file system to use/format in this partition.
In [this example](https://os.mbed.com/docs/mbed-os/v5.15/apis/mbrblockdevice.html) the application sets partition type for FAT (0x83) before formatting the partition with `FATFileSystem::format()`.

In my opinion the user should already know the ID from the doc/code of the filesystem they want to use. So we shouldn't have to link to a Wikipedia page which is both very long and whose URL may change.